src/pages/Scholarships.jsx

```diff
@@ -21,11 +24,51 @@ const useStyles = makeStyles((theme) => ({
 }));
 
 function ScholarshipsPage() {
+  const classes = useStyles();
   const [scholarships, setScholarships] = useState([]);
   const [loading, setLoading] = useState(true);
   const [error, setError] = useState();
-  const [sortField, setSortField] = useState('deadline');
-  const [sortDir, setSortDir] = useState('asc');
+  const history = useHistory();
+  const location = useLocation();
+
+  const validField = Boolean(
+    queryString.parse(location.search).sortField === 'deadline' ||
+      queryString.parse(location.search).sortField === 'amount.min' ||
+      queryString.parse(location.search).sortField === 'amount.max'
+  );
+  const validDir = Boolean(
+    queryString.parse(location.search).sortDir === 'asc' ||
+      queryString.parse(location.search).sortDir === 'desc'
+  );
+
+  let sortField = validField
+    ? queryString.parse(location.search).sortField
+    : DEFAULT_SORT_FIELD;
+  let sortDir = validDir
+    ? queryString.parse(location.search).sortDir
+    : DEFAULT_SORT_DIR;
+
+  const setSort = (val) => {
+    if (val === 'deadlineSoon') {
+      sortField = 'deadline';
+      sortDir = 'asc';
+    } else if (val === 'deadlineLatest') {
+      sortField = 'deadline';
+      sortDir = 'desc';
+    } else if (val === 'amountLow') {
+      sortField = 'amount.min';
+      sortDir = 'asc';
+    } else if (val === 'amountHigh') {
+      sortField = 'amount.max';
+      sortDir = 'desc';
+    }
```

Comment:
I can also do this as a `switch` but it does take more lines to do the same thing and also this logic atleast to me reads more clear than using switch statement.

Reply:
How about storing these in a dictionary like object so you can skip the `if/else` and `switch` statements and just check if the sorting option key is in the dictionary? E.g.
```js
const sortingOptions = {
  'deadlineSoon': {
    field: 'deadline',
    dir: 'asc',
  },
  // etc.
}
```

Reply:
Ok yea nice that is a great idea, right now how I am currently checking if the _key_ exist in the dictionary is by using `.hasOwnPropery(_key_)` is that the correct way to do it? Its working at the moment but im not sure if that's the best way to check.

Reply:
Actually I got it, the best way is to use the `in` key word so 
`if (sortOption in sortingOptions) {}`